rs-matter/src/transport/dedup.rs

```diff
@@ -40,23 +40,29 @@ impl RxCtrState {
     }
 
     /// Receive a message and update Rx State accordingly
-    /// Returns a bool indicating whether the message is a duplicate
-    pub fn recv(&mut self, msg_ctr: u32, is_encrypted: bool) -> bool {
+    /// Returns a bool indicating whether the message updated the internal counter (true)
```

Comment:
These comments describe returning a bool with two meanings:
  - true: if internal counter was updated (state was updated)
  - false: message was a duplicate (state not updated)
  
This double-explanation feels odd even if both seem correct at the same time. Did we consider returning an enum instead? Reading code of a `if post_recv(...)` is not immediately obvious what the bool result is. 

Maybe we find a better name (however I do not have an obvious one :( ). a `post_recv(...) -> ReceivedIndex` with `ReceivedIndex::New` and `ReceivedIndex::Duplicate` may be readable.

Reply:
No this was wrong resolution of a merge conflict hence we ended with two contradictory comments. The internal RX state will only be updated if the message is detected NOT to be a duplicate.

I've adjusted the comment to state so.

I think returning `true` / `false` is good enough here. `false` (= Duplicate) does not result in state change whatsoever.